v2/dtos/requests/device.go

```diff
@@ -0,0 +1,202 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// AddDeviceRequest defines the Request Content for POST Device DTO.
+// This object and its properties correspond to the AddDeviceRequest object in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/AddDeviceRequest
+type AddDeviceRequest struct {
+	common.BaseRequest `json:",inline"`
+	DeviceName         string                             `json:"deviceName" validate:"required"`
+	ServiceName        string                             `json:"serviceName" validate:"required"`
+	ProfileName        string                             `json:"profileName" validate:"required"`
+	AdminState         string                             `json:"adminState"`
+	OperatingState     string                             `json:"operatingState"`
+	Labels             []string                           `json:"labels"`
+	Location           interface{}                        `json:"location"`
+	AutoEvents         []dtos.AutoEvent                   `json:"autoEvents"`
+	Protocols          map[string]dtos.ProtocolProperties `json:"protocols" validate:"required,gt=0"`
```

Comment:
@iain-anderson , Should Protocols be optional?

Reply:
From @iain-anderson 


Hi Lenny, for some reason I'm unable to reply on GitHub to this (although I can to your other comments). In theory you could have a highly specific device service designed to talk to a single instance, for which you would not need Protocols here. But in such a circumstance you can have a Protocol with no ProtocolProperties attached - which is what we do in the examples, eg:


[[DeviceList]]
  Name = 'Bitfields'
  Profile = 'Example-Bitfields'
  Description = 'An Example Bitfields Device'
  [DeviceList.Protocols]
    [DeviceList.Protocols.Bitfields]

So I'm okay with leaving this as required. The vast majority of uses will need it, and those that don't can easily work around.

Reply:
@iain-anderson , currently Device Rest doesn't need ProtocolProperties  and I created a dummy entry. But I am also ok with making it required since as stated vast majority will require it.